src/distilabel/pipeline/base.py

```diff
@@ -243,6 +244,40 @@ def run(
         self._set_runtime_parameters(parameters or {})
         self.dag.validate()
 
+    def dry_run(
+        self,
+        parameters: Optional[Dict[str, Dict[str, Any]]] = None,
+        batch_size: int = 1,
+    ) -> "Distiset":
+        """Do a dry run to test the pipeline runs as expected.
+
+        Running a `Pipeline` in dry run mode will set all the `batch_size` of generator steps
+        to one, and run just with a single batch, effectively running the whole pipeline with
+        a single example. The cache will be set to False.
+
+        Args:
+            parameters: The same parameters variable from `BasePipeline.run`. Defaults to None.
+                Will be passed to the parent method, but with the batch_size of the generator steps
+                fixed to 1.
+            batch_size: The batch size to test the pipeline. Defaults to 1.
+
+        Returns:
+            Will return the `Distiset` as the main run method would do.
+        """
+        self._dry_run = True
+
+        for step_name in self.dag:
+            step = self.dag.get_step(step_name)[STEP_ATTR_NAME]
+            if step.is_generator:
+                self.dag.set_step_attr(step_name, "batch_size", batch_size)
+                # Just in case update the parameters argument
+                if parameters.get(step_name) and parameters[step_name].get(
+                    "batch_size"
+                ):
+                    parameters[step_name]["batch_size"] = batch_size
```

Comment:
Here, we just need `step.batch_size = batch_size`. `Step` instance is an attr of the node of the DAG, but `batch_size` is an attribute of `GeneratorStep`.

Reply:
Not sure of what to update here, the tests work with the update of the runtime parameters, but we have no effect by updating the `step.batch_size`

Reply:
Sorry, I had a brain fart. It's perfectly fine.